site/_posts/2019-12-06-Conformance-Testing-With-Progress-Updates.md

```diff
@@ -0,0 +1,107 @@
+author_avatar: /img/contributors/john-schnake.png
+categories: [kubernetes, sonobuoy, conformance]
+# Tag should match author to drive author pages
+tags: ['Sonobuoy Team']
+---
+
+# Does Testing Kubernetes Conformance Leave You in the Dark? Get Progress Updates as Tests Run
+
+In Sonobuoy 0.15.4, we introduced the ability for plugins to report their plugin's progress to Sonobuoy by using a customizable webhook. Reporting status is incredibly important for long-running, opaque plugins like the `e2e` plugin, which runs the Kubernetes conformance tests.
+
+We're happy to announce that as of Kubernetes 1.17.0, the Kubernetes end-to-end (E2E) test framework will utilize this webhook to provide feedback about how many tests will be run, have been run, and which tests have failed.
+
+This feedback helps you see if tests are failing (and which ones) before waiting for the entire run to finish. It also helps you identify whether tests are hanging or progressing.
+
+## How to Use It
+
+There are two requirements to using this feature for the `e2e` plugin:
+
+ - The conformance image used must correspond to Kubernetes 1.17 or later
+ - Sonobuoy 0.16.5 or later must be used; we added this support prior to 0.17.0 to support Kubernetes prereleases.
+ 
+First, start a run of the `e2e` plugin by running the following command, which kicks off a long-running set of tests:
+
+```
+$ sonobuoy run
+```
+
+Now, you can poll the status by using this command:
+
+```
+$ sonobuoy status --json | jq
+```
+
+After the tests start running, you will start to see output that includes a section like this:
+
+```
+{
+    "plugin": "e2e",
+    "node": "global",
+    "status": "running",
+    "result-status": "",
+    "result-counts": null,
+    "progress": {
+        "name": "e2e",
+        "node": "global",
+        "timestamp": "2019-11-25T17:21:32.5456932Z",
+        "msg": "PASSED [sig-storage] HostPath should give a volume the correct mode [LinuxOnly] [NodeConformance] [Conformance]",
+        "total": 278,
+        "completed": 2
+    }
+}
+```
+
+Voila! Anytime during a run, you can now check in and be more informed about how the run is going. As tests fail, the output will also return an array of strings with the test names in the `failures` field (and the "msg" field just reports the last test finished and its result). For example:
+
+```
+    {
+      ...
+      "progress": {
+        ...
+        "msg": "FAILED [sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should create service with ipv4 cluster ip [Feature:IPv6DualStackAlphaFeature:Phase2]",
+        ...
+        "failures": [
+          "[sig-network] [Feature:IPv6DualStackAlphaFeature] [LinuxOnly] should create service with ipv4 cluster ip [Feature:IPv6DualStackAlphaFeature:Phase2]"
+        ]
+      }
+    },
+```
+
+## Q and A
+
+**Q:** I'm using a new version of Kubernetes but am using an advanced test configuration I store as a YAML file. Can I still get progress updates?
```

Comment:
I think an extra line is needed here to cause the answer to appear on a new line. Same for the other questions below.